### PR TITLE
Modified command line tool descriptions in README.md

### PR DESCRIPTION
**Pull request by michizhou**
Please fill in the fields below to submit a pull request.  The more information
that is provided, the better.

**Fixes issue #159**: Revisit command line tool descriptions in README.md

**Description of the changes being introduced by the pull request**:
Addressed some of @vladimir-v-diaz comments about issues with the README's command line tool descriptions in his gpg support PR review (#150). 

Current behavior/Expected behavior:

Changes that have been implemented (need review):
- Rephrased the general description of `in-toto-run`, shortening its purpose to a single sentence
- Specified that command-line options as what is passed to `in-toto-run` command line tool
- Added link to usage example from demo application for `in-toto-run` to help reduce emphasis on supported command-line options for `in-toto-run`
- Added description for `in-toto-sign` to README

Changes that need to be made (not implemented):
- Rethinking of command hierarchy, e.g. have `in-toto` as single base command with subcommands (`run`, `verify`, `record`, ...), instead of `in-toto-run`, `in-toto-verify`, `in-toto-record`, ...
- Possible addition of more simple usage examples to all commands to reduce emphasis on supported command-line options

**Please verify and check that the pull request fulfills the following
requirements**:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Review by JustinCappos — COMMENTED** [review]
Why is a .DS_Store file added?

**Comment by michizhou** [comment]
@JustinCappos I am not entirely sure. When I made my commit for some reason it added a .DS_Store file to the changes along with those I made to the README. I will get rid of it.

**Comment by JustinCappos** [comment]
Okay, it's also worth you googling to understand where that file came
from.  Definitely look at your commit before you push and be sure it has
exactly what you mean to push and nothing else.

**Comment by michizhou** [comment]
@JustinCappos I understand. I will keep that in mind next time to make sure it doesn't happen again.

File: README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ To learn more about the different rule types, their guarantees and how they are
 #### Carrying out software supply chain steps
 
 ##### in-toto-run
-`in-toto-run` executes the passed command and records the path and hash of the passed *materials* - files before command execution - and *products* - files after command execution and optionally stores them together with the command's *byproducts* (e.g: return value, stdout or stderr) to a link file (`<step-name>.link`), signed with the functionary's key.
+`in-toto-run` generates link metadata for the given command-line option. See the [this simple usage example from the demo application for more details](https://github.com/in-toto/demo#tampering-with-the-software-supply-chain).
```
Comment (SantiagoTorres) [inline README.md]: I don't know how clearer this new message is. I think it's worthy to mention that the command-line option passed is a *command* that will be run (that's why the previous phrasing was a little odd...)

**Comment by SantiagoTorres** [comment]
I updated the target branch to be `develop`. Could you please fix any conflicts? Thanks!

**Comment by michizhou** [comment]
@SantiagoTorres I just fixed the conflicts and merged the results. Let me know if there are any other changes I need to make to the README. Thanks.

**Comment by SantiagoTorres** [comment]
LGTM, thanks @michizhou !

**Comment by aaaaalbert** [comment]
Apologies upfront for my after-the-fact, random spectator type comments...

1. This PR, as merged via 9fb1a4a, introduces a small error in a sentence -- "the this".
2. To what extent is the README's section on *tampering with the software supply chain* a better usage example than, say, the cloning or packaging sections?
3. As discussed elsewhere, I think we should converge to <80 characters per line in Markdown and other plaintext-based docs formats, too.
3. I have a hard time parsing the new explanation for `in-toto-run`, [line 69](https://github.com/in-toto/in-toto/commit/9fb1a4a36699dea9cdab3e3ec519cbe07865207c#diff-04c6e90faac2675aa89e2176d2eec7d8R69):

> `in-toto-run` generates link metadata for the given command-line option and runs it as its own command.

(This should not indicate that `in-toto-run` runs the *metadata* as a command, right?) How about "`in-toto-run` generates link metadata for every build command supplied."?

5. Finally, a small technicality: does your PR fix #159, @michizhou? Then close it, or [provide a commit message that auto-closes it](https://help.github.com/articles/closing-issues-using-keywords/).

**Comment by lukpueh** [comment]
Thanks for your comment, @aaaaalbert! Let's leave #159 open. @adityasaky is currently re-working it. We can close it then when we merge #248.

**Pull request merged**